File: solutions/javascript/collatz-conjecture/1/collatz-conjecture.js

export const steps = (n) => {
     let step = 0
     
     if (n <= 0) throw new Error("Only positive numbers are allowed") 

     while (n > 1){
        if (n % 2 == 0){
          n /= 2
          step += 1      
         }
        else  {
          n = (n * 3) + 1
           step += 1    
        }
     }       
     return step  
};

File: solutions/dart/collatz-conjecture/1/lib/collatz_conjecture.dart
class CollatzConjecture {
    int steps ( n){
        if (n <= 0) throw  new ArgumentError('Only positive integers are allowed'); 
        int steps = 0;

        while (n != 1) {
            if (n % 2 == 0){
                steps += 1;
                n /= 2;
            } 
            else {
              steps += 1;
              n = n * 3 + 1;
            } 
        }
      return steps;
    }
}